src/coreclr/jit/emitarm64.cpp

```diff
@@ -1210,6 +1210,18 @@ void emitter::emitInsSanityCheck(instrDesc* id)
             assert(isValidUimm<2>(emitGetInsSC(id)));  // ii
             break;
 
+        case IF_SVE_CC_2A: // ........xx...... ......mmmmmddddd -- SVE insert SIMD&FP scalar register
+            assert(insOptsScalable(id->idInsOpt()));
+            assert(isVectorRegister(id->idReg1())); // ddddd
+            assert(isVectorRegister(id->idReg2())); // mmmmm
+            break;
+
+        case IF_SVE_CD_2A: // ........xx...... ......mmmmmddddd -- SVE insert general register
+            assert(insOptsScalable(id->idInsOpt()));
+            assert(isVectorRegister(id->idReg1()));  // ddddd
+            assert(isGeneralRegister(id->idReg2())); // mmmmm
```

Comment:
For `SVE_CD_2A` , we should also assert the "width specifier" (<R> field)

![image](https://github.com/dotnet/runtime/assets/12488060/97c40f3f-5c07-4a04-9b08-55287a8d38b3)

Reply:
I'm not sure I can assert this one because it is not passed in as an argument, it is derived from the SVE size. But I have also noticed that it is currently accepting Q size which is invalid and I need to fix this.

Reply:
i think you should be able to do `isValidVectorElemsizeFloat(elemsize)`

Reply:
`elemsize` can be any of the standard four B,H,S,D sizes (as shown in the `<T>` table). The same size field is then reused for the general purpose register, as W or X, with W taking up 3 of the values (as shown in the `<R>` table).
So as long as we assert the B,H,S,D, then the W,X doesn't need checking.